client/src/pages/Join.jsx

```diff
@@ -1,30 +1,25 @@
 import { useState } from "react";
 import { useHistory } from "react-router-dom";
+import useUser from "../api/users";
 import useStyles from "./styles";
 
 function Join() {
   const [error, setError] = useState("");
   const [name, setName] = useState("");
-  const [username, setUsername] = useState("");
+  const { user } = useUser();
   const history = useHistory();
   const classes = useStyles();
 
   const joinLobby = () =>
```

Comment:
joinLobby should be wrapped in a useCallback to prevent it from memoizing the state variables it uses